app/handlers/handlers_connectors.py

```diff
@@ -292,44 +158,49 @@ def twingate_connector_delete(spec, meta, status, namespace, memo, logger, **kwa
         logger.info("Deleting connector %s", connector_id)
         client.connector_delete(connector_id)
 
-    try:
-        # Remove label from pod so its delete handler isn't triggered
-        kapi = kubernetes.client.CoreV1Api()
-        kapi.patch_namespaced_pod(
-            meta.name,
-            namespace,
-            body={"metadata": {"labels": {LABEL_CONNECTOR: None}}},
-        )
-    except kubernetes.client.exceptions.ApiException:
-        logger.exception("Failed to remove label from pod %s", meta.name)
 
+CONNECTOR_RECONCILER_INTERVAL = int(os.environ.get("CONNECTOR_RECONCILER_INTERVAL", "5"))  # fmt: skip
+CONNECTOR_RECONCILER_INIT_DELAY = int(os.environ.get("CONNECTOR_RECONCILER_INIT_DELAY", "5"))  # fmt: skip
 
-@kopf.on.delete("", "v1", "pods", labels={LABEL_CONNECTOR: kopf.PRESENT})
-def twingate_connector_pod_deleted(body, spec, meta, logger, namespace, memo, **_):
-    logger.info("twingate_connector_pod_deleted: %s", body)
 
-    # Annotate the parent connector so that it knows it needs to recreate the pod
-    owner_refs = meta.get("ownerReferences", [])
-    owner = next((o for o in owner_refs if o["kind"] == "TwingateConnector"), None)
-    if not owner:
-        return
+@kopf.timer(
+    "twingateconnector",
+    interval=CONNECTOR_RECONCILER_INTERVAL,
+    initial_delay=CONNECTOR_RECONCILER_INIT_DELAY,
+)
+def twingate_connector_pod_reconciler(
+    body, meta, status, namespace, patch, memo, logger, **_
+):
+    logger.info("twingate_connector_reconciler: %s", body)
+    if not (status and "twingate_connector_create" in status):
+        raise kopf.TemporaryError("TwingateConnector not ready.", delay=1)
 
-    owner_group, owner_version = owner["apiVersion"].split("/")
+    crd = TwingateConnectorCRD(**body)
+    kapi = kubernetes.client.CoreV1Api()
+    k8s_pod = k8s_read_namespaced_pod(namespace, crd.metadata.name, kapi=kapi)
+    image = k8s_pod.spec.containers[0].image if k8s_pod else None
 
-    try:
-        kapi = kubernetes.client.CustomObjectsApi()
-        kapi.patch_namespaced_custom_object(
-            owner_group,
-            owner_version,
-            namespace,
-            "twingateconnectors",
-            owner["name"],
-            {"metadata": {"labels": {LABEL_CONNECTOR_POD_DELETED: "true"}}},
+    if crd.spec.image or not k8s_pod:
+        image = crd.spec.get_image()
+    elif crd.spec.image_policy:
+        now = pendulum.now("UTC").start_of("minute")
+        next_check = pendulum.parse(
```

Comment:
nit: I would prefix this with `_at` to make it obvious that this is a point in time and not a duration which I initially thought 😳